### PR TITLE
Use code-quality-tools

### PR DESCRIPTION
**Pull request by lucasconstantino**
TSLint is in the way to be deprecated in favour of ESLint. We should start migrating.

Ref.:
https://github.com/Microsoft/TypeScript/issues/29288
https://medium.com/palantir/tslint-in-2019-1a144c2317a9
https://eslint.org/blog/2019/01/future-typescript-eslint

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -35,8 +36,7 @@
       "git add README.md"
     ],
     "*.{ts,tsx}": [
-      "prettier-standard",
-      "tslint --fix",
+      "prettier-eslint --write",
```
Comment (dannytce) [inline package.json:39]: Is this one actually needed? Isn't it easier to go simply with
```
eslint --fix
prettier --write
```
? :)

File: package.json [hunk 2]
```diff
@@ -143,6 +147,8 @@
     "chance": "^1.0.18",
     "dotenv": "^7.0.0",
     "dotenv-webpack": "^1.7.0",
+    "eslint": "^5.16.0",
+    "eslint-plugin-react": "^7.12.4",
```
Comment (dannytce) [inline package.json:151]: You don't need to have this one, it's already a part of `@strv/eslint-config-react`

**Review by dannytce — APPROVED** [review]
Overall looks very good 👍 I found a few nitpicks, see the comments :)

And I have one more request: could you also add `stylelint` to the project? We have a preset of rules in `code-quality-tools` repository too :)

File: package.json [hunk 3]
```diff
@@ -35,8 +36,7 @@
       "git add README.md"
     ],
     "*.{ts,tsx}": [
-      "prettier-standard",
-      "tslint --fix",
+      "prettier-eslint --write",
       "stylelint --fix",
```
Comment (dannytce) [inline package.json]: It's unrelated to this PR, but let's remove --fix flag from stylelint - it won't work with styled-components

File: package.json (hunk 1 above)
Comment (lucasconstantino) [reply package.json:39]: @dannytce prettier is known for having conflicting issues when run after ESLint, which would ultimately mean the code would change twice and end-up on the prettier version.

`prettier-eslint` is basically a shortcut to run prettier, but respecting any conflicting rules from ESLint config.

File: package.json (hunk 2 above)
Comment (lucasconstantino) [reply package.json:151]: @dannytce `@strv/eslint-config-react` was installing version `7.11.1` of `eslint-plugin-react`, which lacked some configuration for a rule a honestly don't remember which. Given the semantic versioning constraint, I expected both to be equivalent, but that did no happen to me, and I didn't dive deeper.

I'll give a try again on removing that and see how it goes ;)

File: package.json (hunk 3 above)
Comment (lucasconstantino) [reply package.json]: @dannytce this `lint-staged` is meant to actually fix the code. If we remove `--fix`, we might as well just remove `stylelint` script altogether, right?

Ps.: I didn't change anything related to `stylelint`, but now that I'm looking at it, I'll add this linter to the `qa` script - which combines all code validation tools.

File: package.json (hunk 1 above)
Comment (dannytce) [reply package.json:39]: So either you go with `prettier-eslint` or extending your ESLint config with `eslint-config-prettier`, right? :)

File: package.json (hunk 3 above)
Comment (dannytce) [reply package.json]: @lucasconstantino I think that `lint-staged` is not just about fixing the code, but mostly not letting you commit 💩code :) Any autofixes there are just additional bonuses. 

The problem with `stylelint --fix` is that it's not working well with styled-componenets at the moment. It basically breaks the whole styles. But I think it's a valid to throw an error (and not commit) if styles are broken anyhow.

Last thing, you can use [our stylelint config](https://github.com/strvcom/code-quality-tools/tree/master/packages/stylelint-config-styled-components) to get rid of a few dependencies :)

**Pull request merged**